nistats/reporting/glm_reporter.py

```diff
@@ -0,0 +1,945 @@
+
+Functions
+---------
+
+make_glm_report(model, contrasts):
+    Creates an HTMLDocument Object which can be viewed or saved as a report.
+
+"""
+
+import io
+import string
+import os
+import warnings
+
+from collections import OrderedDict
+
+try:
+    from collections.abc import Iterable
+    from html import escape
+    from urllib.parse import quote
+except ImportError:  # Python2
+    from collections import Iterable
+    from cgi import escape
+    from urllib import quote
+
+import numpy as np
+import pandas as pd
+
+from matplotlib import pyplot as plt
+from nilearn.plotting import (plot_glass_brain,
+                              plot_roi,
+                              plot_stat_map,
+                              )
+from nilearn.plotting.img_plotting import MNI152TEMPLATE
+from nilearn.plotting.js_plotting_utils import HTMLDocument
+
+import nistats
+from nistats.reporting import (plot_contrast_matrix,
+                               plot_design_matrix,
+                               get_clusters_table,
+                               )
+from nistats.thresholding import map_threshold
+
+
+HTML_TEMPLATE_ROOT_PATH = os.path.join(os.path.dirname(__file__),
+                                       'glm_reporter_templates')
+
+
+def make_glm_report(model,
+                    contrasts,
+                    title=None,
+                    bg_img=MNI152TEMPLATE,
+                    threshold=3.09,
+                    alpha=0.01,
+                    cluster_threshold=0,
+                    height_control='fpr',
+                    min_distance=8.,
+                    plot_type='slice',
+                    display_mode=None,
+                    report_dims=(1600, 800),
+                    ):
+    """ Returns HTMLDocument object
+    for a report which shows all important aspects of a fitted GLM.
+    The object can be opened in a browser, displayed in a notebook,
+    or saved to disk as a standalone HTML file.
+    
+    Examples:
+        report = make_glm_report(model, contrasts)
+        report.open_in_browser()
+        report.save_as_html(destination_path)
+    
+    Parameters
+    ----------
+    model: FirstLevelModel or SecondLevelModel object
+        A fitted first or second level model object.
+        Must have the computed design matrix(ces).
+        
+    contrasts: Dict[string, ndarray] or String or List[String] or ndarray or
+        List[ndarray]
+        
+        Contrasts information for a first or second level model.
+        
+        Example:
+        
+            Dict of contrast names and coefficients,
+            or list of contrast names
+            or list of contrast coefficients
+            or contrast name
+            or contrast coefficient
+            
+            Each contrast name must be a string.
+            Each contrast coefficient must be a list or numpy array of ints.
+        
+        Corresponds to the ``contrast_def`` for the FirstLevelModel [1]_
```

Comment:
i think this is confusing as the firstlevelmodel compute_contrast doesn't have the same interface

Reply:
What do you suggest?
I have tried to be specific for that reason, but you have a better docstring, we can change this.

Reply:
> What do you suggest?

just avoiding "corresponds to"
e.g. "contrasts are passed to `flm.compute_contrast`"